sdk/src/main/java/io/customer/sdk/queue/QueueRunner.kt

```diff
@@ -40,13 +47,27 @@ internal class QueueRunnerImpl(
     private suspend fun identifyProfile(task: QueueTask): QueueRunTaskResult {
         val taskData: IdentifyProfileQueueTaskData = jsonAdapter.fromJson(task.data)
 
-        return cioHttpClient.identifyProfile(taskData.identifier, taskData.attributes)
+        return cioHttpClient.identifyProfile(taskData.identifier, taskData.attributes).apply {
+            if (this.isSuccess) {
+                hooksManager.onHookUpdate(
+                    hook = ModuleHook.ProfileIdentifiedHook(taskData.identifier)
+                )
+            }
+        }
```

Comment:
Is there a reason why module hook events are being called here in the background queue execution instead of when `CustomerIO.identify()` is called?

Reply:
Good point, two clarifications needed on it, 

- Where do we notify hook, I agree, I would rather have it repositories rather than Queue, it would make more sense and be more consistant. 
-  When do we notify? I am currently going for when the queue has run the task to be consistent, otherwise, behaviour can be a bit different, because imagine on fly we dont see a person being notified for the next 30 secs, but they are getting in-app messages on phone since we already set the `user-token` on Gist.

Reply:
From your reply ^^^, it seems to me that you are planning on moving the notify hooks code from here in the `QueueRunner` to the repositories? I can't tell from your comment what your action item is. 

I agree that the repositories would be a better place for this code.

Reply:
Looks like the change was made. Looks good!